.github/workflows/main.yml

```diff
@@ -1,40 +1,83 @@
+# actions can be run locally using act and docker, on Fedora 37 also with podman, using:
+# https://github.com/nektos/act
+# sudo dnf install -y act-cli podman-docker
+# act --bind --container-daemon-socket $XDG_RUNTIME_DIR/podman/podman.sock -W .github/workflows/main.yml
+
 name: Unit tests
 
 on: [push, pull_request]
+env:
+  PYTHONWARNINGS: "ignore"
+  PIP_ROOT_USER_ACTION: "ignore"  # For local testing using act-cli (see notes above)
 
 jobs:
   test:
     strategy:
+      max-parallel: 1
```

Comment:
They could, but for now, I'd prefer the run then in serial for two reasons:

When testing the action locally using `act-cli.rpm` / https://github.com/nektos/act, act also runs all jobs in parallel and if a job fails, the error is scrolled out of sight very quickly.

When testing with parallel jobs using `act` I also now run into this error, which I didn't see with execution in serial:
```py
| + /root/.local/bin/tox
| py38: internal error
| Traceback (most recent call last):
|   File "/root/.local/lib/python3.8/site-packages/tox/session/cmd/run/single.py", line 45, in _evaluate
|     tox_env.setup()
|   File "/root/.local/lib/python3.8/site-packages/tox/tox_env/api.py", line 249, in setup
|     self._setup_env()
|   File "/root/.local/lib/python3.8/site-packages/tox/tox_env/python/runner.py", line 106, in _setup_env
|     super()._setup_env()
|   File "/root/.local/lib/python3.8/site-packages/tox/tox_env/python/api.py", line 198, in _setup_env
|     self.ensure_python_env()
|   File "/root/.local/lib/python3.8/site-packages/tox/tox_env/python/api.py", line 205, in ensure_python_env
|     self.create_python_env()
|   File "/root/.local/lib/python3.8/site-packages/tox/tox_env/python/virtual_env/api.py", line 129, in create_python_env
|     self.session.run()
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/run/session.py", line 44, in run
|     self._create()
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/run/session.py", line 51, in _create
|     self.creator.run()
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/create/creator.py", line 151, in run
|     self.create()
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/via_global_self_do.py", line 95, in create
|     src.run(self, self.symlinks)
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/ref.py", line 153, in run
|     method(self.src, dest)
|   File "/root/.local/lib/python3.8/site-packages/virtualenv/util/path/_sync.py", line 31, in symlink
|     dest.symlink_to(src, target_is_directory=src.is_dir())
|   File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/pathlib.py", line 1384, in symlink_to
|     self._accessor.symlink(target, self, target_is_directory)
|   File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/pathlib.py", line 446, in symlink
|     return os.symlink(a, b)
| FileExistsError: [Errno 17] File exists: '/opt/hostedtoolcache/Python/3.8.16/x64/bin/python' -> '/home/bkaindl/git/github/python-libs/.tox/py38/bin/python'
```

Lastly, if there is any reason for the CI to fail, it likely fails during the 1st full run of the workflow, and by not starting them all in parallel we reduce our time budget for gitlib runners. If we overflow the limit of 2000 minutes of execution, we'd have no execution time for the rest of the month (or we've to switch to the GitHub Enterprice Cloud product...):

https://medium.com/@milwojarski/release-the-limits-of-github-actions-with-self-hosted-runners-9ecdb6419934
https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

Reply:
I recommend reducing the number of tested python version: 2.7, 3.6 and 3.11 instead of testing them all like this. at least during development. Furthermore I think if tox is configured to set up all python versions there's no need to do the testing matrix using github action, which separates the actual testing of the library on all version from the development of the pipeline, which means you don't have to worry about act interference

Reply:
I've pushed my I hope final big update now. It moves all the pytest and pylint into tox!

This reduced the amount of calls and the overhead extremely, it's really fast with tox!

The execution time is now 27s to 49s per job, so it's a no-brainer to run them all now, all run in parallel, I'm very happy and excited that I've tox made this possible!

Please review! I'll put the documentation into README.md, I hope this PR is ready then!

Afterwards I already have a number other PRs to fix the remaining Py3 issues in the queue.